### ISSUE TITLE
RSX: Unsupported texture formats

### ISSUE THREAD

**Issue by SakataGintokiYT** [RPCS3/rpcs3#1527]
{rsx::thread} RSX: Unknown/illegal instruction: 0x17 (forced unit 0)
{rsx::thread} RSX: Unknown/illegal instruction: 0x17 (forced unit 0)
_____________________________________________________________________
SOULCALIBUR IV
{rsx::thread} Exception: Wrong format 158
(in file Emu\RSX\Common\TextureUtils.cpp:301, in function upload_placed_texture)
_____________________________________________________________________
Marvel vs Capcom 2
{rsx::thread} Exception: Wrong format 139
(in file Emu\RSX\Common\TextureUtils.cpp:301, in function upload_placed_texture)
_____________________________________________________________________
-Marvel vs Capcom 2 [NPEB00087]
-SOULCALIBUR IV [NPUB90142]
_____________________________________________________________________
<img src="http://www.fotoszok.pl/upload/0fa81cc7.png" alt="2015-12-13 20_52_51-xenia" style="max-width:50%;">

##### [NPEB00087]: http://www.emunewz.net/forum/showthread.php?tid=173862&highlight=marvel

**Comment by MyaniPT** [comment]
Game?
Log?

**Comment by SakataGintokiYT** [comment]
Update

**Comment by vlj** [comment]
Do these game run without hack ? At least Marvel vs Capcom.

**Comment by SakataGintokiYT** [comment]
Yes working without hack Marvel vs Capcom [ log and screenshot] tested on master build

**Comment by kd-11** [comment]
There seems to be some unimplemented texture formats right now. That's the issue here, same as with god of war collection. The title should probably be better to direct developers.

**Comment by MyaniPT** [comment]
Also happens in NPUB31180

**Comment by SakataGintokiYT** [comment]
Marvel vs Capcom 2 [NPEB00087] Fixed ( Wrong format 139 / TextureUtils.cpp:301)
<img src="http://www.fotoszok.pl/upload/c6658cc1.jpg" alt="Fotoszok.pl" />
_____________________________________________________________________
Ice Age: Dawn of the Dinosaurs  (New issue  Wrong format 151 / TextureUtils.cpp:340 )
{rsx::thread} Exception: Wrong format 151
(in file C:\rpcs3\rpcs3\Emu\RSX\Common\TextureUtils.cpp:340, in function upload_placed_texture)
_____________________________________________________________________

**Comment by kd-11** [comment]
@SakataGintokiYT The PR has been updated. You can test once the appveyor build is ready.

**Comment by SakataGintokiYT** [comment]
@kd-11 
Fixed TextureUtils.cpp:340
<img src="http://www.fotoszok.pl/upload/c3973565.jpg" alt="Fotoszok.pl" />

**Comment by SakataGintokiYT** [comment]
@kd-11 
Immutable textures cannot be reused!-31de4a9f

Ice Age 3: Dawn of the Dinosaurs
New issue:
{rsx::thread} Exception: Wrong format 151
(in file C:\rpcs3\rpcs3\Emu\RSX\Common\TextureUtils.cpp:337, in function upload_placed_texture)
-------------------------------------------------------------------------------------------------------
Marvel vs. Capcom 2
New issue:
{rsx::thread} Exception: Wrong format 139
(in file C:\rpcs3\rpcs3\Emu\RSX\Common\TextureUtils.cpp:337, in function upload_placed_texture)

**Comment by SakataGintokiYT** [comment]
@kd-11 
Castlevania: Lords of Shadow 2
{rsx::thread} Exception: Wrong format 156
(in file C:\rpcs3\rpcs3\Emu\RSX\Common\TextureUtils.cpp:174, in function get_subresources_layout)

**Comment by kd-11** [comment]
@SakataGintokiYT Please retest castlevania. It might still crash, but the emulator has had alot of progress recently. Also, is that vulkan backend or OpenGL?

**Comment by SakataGintokiYT** [comment]
@kd-11 
[27a8c074]
*DX12 - The same issue 
*Vulkan / OpenGL  The same issue  +
{rsx::thread} RSX: Surface color buffer: Unsupported surface color format (0xa)

**Comment by SakataGintokiYT** [comment]
<img src="http://www.fotoszok.pl/upload/4b1de409.jpg" alt="Fotoszok.pl" />

**Comment by tambry** [comment]
That's CELL_GCM_TEXTURE_X32_FLOAT. Shouldn't be too difficult to implement, probably.

**Comment by tambry** [comment]
I'm unable to start Castlevania [NPEB01434] due to it trying to display a error dialog about being "corrupt". Also I encountered a wxWidgets assert, which I fixed, but after which the message dialog is unresponsive and doesn't render properly. I stripped down the message dialog code to bare minimum, but I'm unsure what's wrong.w

**Comment by raven02** [comment]
Castlevania , no longer bootup in 0.0.0.9

{PPU[0x3] Thread (main_thread) [0x00cd9d60]} class std::runtime_error thrown: Verification failed: vm::dealloc(img->segs.addr(), vm::main)
(in file C:\rpcs3\rpcs3\Emu\Cell\Modules\sys_spu_.cpp:62)

**Comment by raven02** [comment]
i think this one can be closed as i didn't see any more unsupported format in log now.
